src/transformers/modeling_utils.py

```diff
@@ -1293,6 +1597,10 @@ def from_pretrained(cls, pretrained_model_name_or_path: Optional[Union[str, os.P
         else:
             model_kwargs = kwargs
 
+        # This variable will flag if we're loading a sharded checkpoint. In this case the archive file is just the
+        # index of the files.
+        is_sharded = False
+        sharded_metadata = None
```

Comment:
We are in `from_pretrained` and the main diff now, this is the part that needs the most attention :-)